spectool/jdutils.ts

```diff
@@ -66,7 +67,7 @@ export class SpecSymbolResolver {
 
     constructor(
         private readonly spec: jdspec.ServiceSpec,
-        private readonly role2spec: (role: string) => jdspec.ServiceSpec,
+        private readonly role2spec: (role: string) => { spec: jdspec.ServiceSpec, client: boolean},
```

Comment:
considering client is the default scenario, refactor ``client: boolean`` to ``server?: boolean`` so that missing/falsy value means client.